src/Model.php

```diff
@@ -237,6 +237,19 @@ public function __construct($persistence = null, $defaults = [])
         }
     }
 
+    public function __clone()
+    {
+        // we need to clone some of the elements
+        if ($this->elements) {
+            foreach ($this->elements as $id => $el) {
+                if ($el instanceof Join) {
+                    $this->elements[$id] = clone $el;
+                    $el->owner = $this;
```

Comment:
I guess there is bug. $el variable will be lost after foreach anyway. Probably $this->elements[$id]->owner should be there instead of $el->owner.